RoaringBitmap/src/main/java/org/roaringbitmap/ArrayContainer.java

```diff
@@ -359,6 +359,17 @@ public void deserialize(DataInput in) throws IOException {
     }
   }
 
+  @Override
+  public void deserialize(ByteBuffer byteBuffer) throws IOException {
+    this.cardinality = 0xFFFF & byteBuffer.getChar();
```

Comment:
Why does this have a cardinality but the `BitmapContainer` doesn't? I don't agree with these methods because they seem specific to the ART high/low container, but contributors would be tempted to use them with `RoaringArray`

Reply:
I am confused by the `writeArray`. I initially want to use the writeArray. But I found that the `writeArray` implementation logic is different from the corresponding `serialize(DataOutput )`, see `ArrayContainer` implementation for example. I think the writeArray's implementation should have the same logic with the serialize(DataOutput ) version.  These's no corresponding deserialization version to `writeArray`. That's why I implemented the serialize/deserialize (ByteBuffer)